### PR TITLE
chore(tests): Add support for proptest to lookup types

### PR DESCRIPTION
**Pull request by bruceg**
This pulls in the latest rev of the VRL crate to get proptest support for the types underlying the lookup types.

### CODE REVIEW

File: lib/vector-core/src/fanout.rs [hunk 1]
```diff
@@ -74,7 +74,7 @@ impl Fanout {
 
     fn remove(&mut self, id: &ComponentKey) {
         assert!(
-            self.senders.remove(id).is_some(),
+            self.senders.shift_remove(id).is_some(),
```
Comment (pront) [inline lib/vector-core/src/fanout.rs:77]: Is this related to this PR?

**Review by pront — APPROVED** [review]
Lookup related additions look good

**Review by dsmith3197 — APPROVED** [review]

File: lib/vector-core/src/fanout.rs (hunk 1 above)
Comment (dsmith3197) [reply lib/vector-core/src/fanout.rs:77]: This is a result of bumping the `indexmap` version.

**Review by pront — APPROVED** [review]
Thanks for updating the examples

**Review by hestonhoffman — APPROVED** [review]
